### PR TITLE
Payment system

### PR DESCRIPTION
**Pull request by miikebar**

### CODE REVIEW

File: app/src/components/wallet/WalletTransactionHistoryList.tsx [hunk 1]
```diff
@@ -65,6 +65,8 @@ export const WalletTransactionHistoryList: React.FC = () => {
       contentContainerStyle={styles.contentContainer}
       data={mockData}
       renderItem={renderItem}
+      // TODO: use item id
+      keyExtractor={(item, index) => index.toString()}
```
Comment (KalitaKonrad) [inline app/src/components/wallet/WalletTransactionHistoryList.tsx:69]: Why not using `item.id` right away?

File: app/src/components/wallet/WalletTransactionHistoryList.tsx [hunk 2]
```diff
@@ -65,6 +65,8 @@ export const WalletTransactionHistoryList: React.FC = () => {
       contentContainerStyle={styles.contentContainer}
       data={mockData}
       renderItem={renderItem}
+      // TODO: use item id
```
Comment (KalitaKonrad) [inline app/src/components/wallet/WalletTransactionHistoryList.tsx]: Can `item.id` be used now?

File: app/src/screens/wallet/WalletPaymentMethodScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { StackScreenProps } from '@react-navigation/stack';
+import React, { useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+// @ts-ignore
```
Comment (KalitaKonrad) [inline app/src/screens/wallet/WalletPaymentMethodScreen.tsx]: Remove ts-ignore.

File: app/src/screens/wallet/WalletPaymentMethodScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        preparePaymentMethodPayload(card)
+      );
+      const { id } = response.data;
+      await updatePaymentMethod(id, { throwOnError: true });
+      navigation.navigate('Wallet');
+    } catch (error) {
+      alert(
+        'Podczas zapisywania metody płatności wystąpił błąd. Spróbuj ponownie lub skontaktuj się z supportem.'
+      );
+      console.log(error.response);
```
Comment (KalitaKonrad) [inline app/src/screens/wallet/WalletPaymentMethodScreen.tsx:66]: Remove console.log

File: app/src/screens/wallet/WalletPaymentMethodScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    } finally {
+      setPending(false);
+    }
+  };
+
+  return (
+    <Container>
+      <PaddedInputScrollView>
```
Comment (KalitaKonrad) [inline app/src/screens/wallet/WalletPaymentMethodScreen.tsx:92]: Didn't we swap out this library? I guess the name of the component stayed the same :)

File: app/src/screens/wallet/WalletScreen.tsx [hunk 6]
```diff
@@ -10,20 +10,45 @@ import { NumberSelector } from '../../components/shared/NumberSelector';
...
   const [topUpAmount, setTopUpAmount] = useState(0);
+  const [topUp] = useChargeWalletMutation();
+  const [isPending, setPending] = useState(false);
+
+  const onTopUpPress = async () => {
+    setPending(true);
+
+    try {
+      const response = await topUp(topUpAmount, { throwOnError: true });
+      console.log(response);
```
Comment (KalitaKonrad) [inline app/src/screens/wallet/WalletScreen.tsx]: remove console.log

File: app/src/types/payments/payment-method.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
+declare module Stripe {
```
Comment (KalitaKonrad) [inline app/src/types/payments/payment-method.ts:1]: What is the point of declaration as a module? To group them and "mark" as connected to stripe?

**Review by KalitaKonrad — APPROVED** [review]

File: app/src/types/payments/payment-method.ts (hunk 7 above)
Comment (miikebar) [reply app/src/types/payments/payment-method.ts:1]: Yes

**Review by SikoraKam — APPROVED** [review]

**Pull request merged**